### PR TITLE
[Bugfix][MiniMax-H3] Stabilize keyframe VAE encoding

### LINKED ISSUE #7035: [Bug]: Occasional,  Nightly / CI failed - tests/e2e/accuracy/minimax_h3/test_minimax_h3_i2va_ref2va_similarity.py::test_minimax_h3_i2va_matches_official_reference - AssertionError: SSIM below threshold: got 0.967463
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
ci env
```

</details>

### Your code version

<details>
<summary>The commit id or version of vllm</summary>

```text
(not found in Buildkite step log)
```
</details>
<details>
<summary>The commit id or version of vllm-omni</summary>

```text
ec22c7ac06ad
```
</details>

### 🐛 Describe the bug

**Buildkite build:** https://buildkite.com/vllm/vllm-omni/builds/14599
**Buildkite step:** [:full_moon: Diffusion X2V · MiniMax H3 I2VA/Ref2VA Accuracy Test](https://buildkite.com/vllm/vllm-omni/builds/14599#01a066b9-e615-4e54-a67d-8fa8628a9574)

**Failure kind:** pytest FAILED
**Test node:** `tests/e2e/accuracy/minimax_h3/test_minimax_h3_i2va_ref2va_similarity.py::test_minimax_h3_i2va_matches_official_reference`

**Log reason:**
AssertionError: SSIM below threshold: got 0.967463, expected >= 0.970000. online=/workspace/build/buildkite/tests/e2e/accuracy/artifacts/minimax_h3_i2va/vllm_omni.mp4 offline=/workspace/build/buildkite/tests/e2e/accuracy/artifacts/minimax_h3_i2va/reference.mp4

**Analysis:**
Category: assertion failure. Expected vs actual mismatch — check test logic, edge cases, and data setup.

**Error log excerpt:**

```text
accuracy_artifact_root = PosixPath('/workspace/build/buildkite/tests/e2e/accuracy/artifacts')



    @hardware_test(res={"cuda": "H100"}, num_cards=4)

    def test_minimax_h3_i2va_matches_official_reference(

        accuracy_artifact_root: Path,

    ) -> None:

        if not torch.cuda.is_available():

            pytest.skip("MiniMax H3 I2VA accuracy test requires CUDA.")

    

        probe_binary("ffmpeg")

        probe_binary("ffprobe")

        output_dir = reset_artifact_dir(accuracy_artifact_root / "minimax_h3_i2va")

        reference_path = _download_reference_asset(

            output_dir,

         

...(truncated for GitHub URL length; see HTML report for full log)
```

---
*Generated from a nightly HTML report. Redact secrets before submitting; complete the checkboxes on GitHub.*

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by david6666666**
Fixes #7035

## Summary
- Pin the MiniMax-H3 GPU keyframe VAE encode to deterministic cuDNN algorithms with TF32 enabled.
- Scope the setting to `encode_image` and restore the caller's backend state on exit.
- Cover the cuDNN settings, state restoration, and the `encode_image` call path, including encoder failures.

## Root cause
The Buildkite failure is reproducible as an environment-sensitive accuracy failure, but no causal code commit is isolated. Build 14631 at `10701e18` passed the same I2VA gate (SSIM 0.972555, PSNR 34.299243 dB), while Build 14679 at `5d7fdf93` and Build 14708 at `039808e0` failed with the same output (SSIM 0.971247, PSNR 33.627597 dB). The runs use the same H100 driver/CUDA stack but different H100 card groups. The FP32 3D-CNN keyframe encode inherited ambient cuDNN algorithm selection; small reduction differences are amplified by the 50-step denoiser.

## Validation
- 133 MiniMax-H3 contract tests passed. Removing the production encode context makes both new integration cases fail.
- Latest `main` (`38abc0ba`) on 4x NVIDIA B300 with the Buildkite topology and arguments: SSIM 0.974140, PSNR 35.375611 dB.
- Pre-commit hooks passed, including ruff, mypy 3.10, SPDX, import, and CI-mark checks.

Accuracy thresholds and non-keyframe paths are unchanged.

### CODE REVIEW

**Comment by david6666666** [comment]
image

File: tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py [hunk 1]
```diff
@@ -1550,6 +1550,43 @@ def __init__(self):
...
+        cudnn.deterministic,
+        cudnn.allow_tf32,
+    )
+    try:
+        cudnn.enabled = False
+        cudnn.benchmark = True
+        cudnn.deterministic = False
+        cudnn.allow_tf32 = False
+
+        with _minimax_h3_keyframe_encode_context(torch.device("cuda")):
```
Comment (david6666666) [inline tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py:1572]: Can we also test that `encode_image` enters this context? This test stays green if the `with` at the call site is removed.

Comment (david6666666) [reply tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py:1572]: Fixed in 91634e966. Added CPU coverage through `encode_image` for success and failure. Removing the production `with` fails both cases; all 133 contract tests pass.
